Gordon360/Services/RecIM/ParticipantService.cs

```diff
@@ -105,29 +105,78 @@ public IEnumerable<ParticipantStatusExtendedViewModel> GetParticipantStatusHisto
-                                    ID = t.ID,
-                                    Activity = context.Activity.FirstOrDefault(a => a.ID == t.ActivityID),
-                                    Name = t.Name,
-                                    Status = context.TeamStatus
-                                                .FirstOrDefault(ts => ts.ID == t.StatusID)
-                                                .Description,
-                                    Logo = t.Logo,
-                                    TeamRecord = context.SeriesTeam
-                                        .Include(st => st.Team)
-                                        .Where(st => st.TeamID == t.ID)
-                                        .Select(st => (TeamRecordViewModel)st)
-                                
-                                });
+        var teams = context.ParticipantTeam
+                 .Where(pt => pt.ParticipantUsername == username && pt.RoleTypeID != 0 && pt.RoleTypeID != 2)
+                     .Join(context.Team.Where(t => t.StatusID != 0),
+                         pt => pt.TeamID,
+                         t => t.ID,
+                         (pt, t) => new TeamExtendedViewModel
+                         {
+                             ID = t.ID,
+                             Activity = context.Activity.FirstOrDefault(a => a.ID == t.ActivityID),
+                             Name = t.Name,
+                             Status = context.TeamStatus
+                                         .FirstOrDefault(ts => ts.ID == t.StatusID)
+                                         .Description,
+                             Logo = t.Logo,
+                             TeamRecord = context.SeriesTeam
+                                 .Include(st => st.Team)
+                                 .Where(st => st.TeamID == t.ID)
+                                 .Select(st => (TeamRecordViewModel)st)
+
+                         });
         return teams;
     }
 
+    public IEnumerable<MatchExtendedViewModel> GetParticipantMatches(string username)
+    {
+        var matches = context.ParticipantTeam
+            .Where(pt => pt.ParticipantUsername == username && pt.RoleTypeID != 0 && pt.RoleTypeID != 2)
+                .Join(context.MatchTeam.Where(mt => mt.StatusID != 0 && mt.Match.StatusID != 0)
+                                        .Include(mt => mt.Match)
+                                            .ThenInclude(m => m.Series)
+                                                .ThenInclude(s => s.Activity)
+                                        .Include(mt => mt.Match)
+                                            .ThenInclude(m => m.MatchTeam)
+                                                .ThenInclude(mt => mt.Match)
+                                        .Include(mt => mt.Match)
+                                            .ThenInclude(m => m.MatchTeam)
+                                                .ThenInclude(mt => mt.Status),
```

Comment:
I'm fairly certain you don't need to include an entity more than once:
```suggestion
                                        .Include(mt => mt.Match)
                                            .ThenInclude(m => m.Series)
                                                .ThenInclude(s => s.Activity)
                                            .ThenInclude(m => m.MatchTeam)
                                                .ThenInclude(mt => mt.Match)
                                                .ThenInclude(mt => mt.Status),
```

Reply:
True, didn't notice the overlap during development

Reply:
Wait no nevermind, (I don't have access to VMs right now). 
The way those .ThenIncludes reads is that is
Match -> Series -> Activity -> MatchTeam 
Which makes no sense. I have to find it via Match teach time.